wasm/datatypes/stack.py

```diff
@@ -0,0 +1,86 @@
+from typing import (
+    Generic,
+    Iterable,
+    List,
+    NamedTuple,
+    Tuple,
+    TypeVar,
+)
+import uuid
+
+from wasm.typing import (
+    TValue,
+)
+
+from .indices import (
+    LabelIdx,
+)
+from .instructions import (
+    InstructionSequence,
+)
+from .module import (
+    ModuleInstance,
+)
+
+TStackItem = TypeVar('TStackItem')
+
+
+class BaseStack(Generic[TStackItem]):
+    _stack: List[TStackItem]
+
+    def __init__(self) -> None:
+        self._stack = []
+
+    def __len__(self) -> int:
+        return len(self._stack)
+
+    def __bool__(self) -> bool:
+        return bool(self._stack)
+
+    def __iter__(self) -> Iterable[TStackItem]:
+        return iter(self._stack)
+
+    def pop(self) -> TStackItem:
+        return self._stack.pop()
+
+    def pop2(self) -> Tuple[TStackItem, TStackItem]:
+        return self._stack.pop(), self._stack.pop()
+
+    def pop3(self) -> Tuple[TStackItem, TStackItem, TStackItem]:
+        return self._stack.pop(), self._stack.pop(), self._stack.pop()
+
+    def push(self, value: TStackItem) -> None:
+        self._stack.append(value)
+
+    def peek(self) -> TStackItem:
+        return self._stack[-1]
+
+
+class ValueStack(BaseStack[TValue]):
+    pass
+
+
+class Frame(NamedTuple):
+    id: uuid.UUID
+    module: ModuleInstance
+    locals: List[TValue]
+    instructions: InstructionSequence
+    arity: int
+    height: int
+
+
+class FrameStack(BaseStack[Frame]):
```

Comment:
Is there a reason why this needs to be a class? A simple alias `FrameStack = BaseStack[Frame]` should do it unless there's something I'm not aware of.

Reply:
Ah, I see the `LabelStack` further down adds a method, so maybe the class based syntax is just for consistency then. Feel free to ignore.

Reply:
I had similar thoughts.

The `peek()`, `pop2()` and `pop3()` APIs are all used and losing them isn't a big deal.  I'll ponder a bit but I'm leaning towards keeping them as formal data structures.